src/ui/common/src/components/pages/artifact/id/index.tsx

```diff
@@ -205,10 +205,69 @@ const ArtifactDetailsPage: React.FC<ArtifactDetailsPageProps> = ({
   const inputs = mapOperators([artifact.from]);
   const outputs = mapOperators(artifact.to ? artifact.to : []);
 
+  let upstream_pending = false;
+  inputs.some((operator) => {
+    const operator_pending =
+      operator.result.exec_state.status === ExecutionStatus.Pending;
+    if (operator_pending) {
+      upstream_pending = operator_pending;
+    }
+    return operator_pending;
+  });
+
   const artifactStatus = artifact?.result?.exec_state?.status;
   const previewAvailable =
     artifactStatus && artifactStatus !== ExecutionStatus.Canceled;
 
+  let preview = (
```

Comment:
This file is starting to get a little long. Could you make a components folder at 
```artifact/id/components``` and turn the Preview into it's own component please?

Doesn't have to be now, you can add a todo and address in a future PR.

Reply:
Good suggestion! Will do